trunk/SUAVE/Input_Output/OpenVSP/vsp_read_wing.py

```diff
@@ -0,0 +1,238 @@
+	
+	total_chord      = vsp.GetParmVal(wing_id, 'Root_Chord', 'XSec_1')	* units_factor
+	total_proj_span  = vsp.GetParmVal(wing_id, 'TotalProjectedSpan', 'WingGeom')  * units_factor
+	span_sum         = 0.				# Non-projected.
+	proj_span_sum    = 0.				# Projected.
+	segment_spans    = [None] * (segment_num) 	# Non-projected.
+	segment_dihedral = [None] * (segment_num)
+	segment_sweeps_quarter_chord = [None] * (segment_num)
+	
+	# Check for wing segment *inside* fuselage, then skip XSec_0 to start at first exposed segment.
+	if vsp.GetParmVal(wing_id, 'Root_Chord', 'XSec_0') == 1.:
+		start = 1
+	else:
+		start = 0
+	
+	# -------------
+	# Wing segments
+	# -------------		
+	
+	# Convert VSP XSecs to SUAVE segments. (Wing segments are defined by outboard sections in VSP, but inboard sections in SUAVE.) 
+	for i in range(start, segment_num+1):		
+		segment = SUAVE.Components.Wings.Segment()
+		segment.tag                   = 'Section_' + str(i)
+		thick_cord                    = vsp.GetParmVal(wing_id, 'ThickChord', 'XSecCurve_' + str(i-1))
+		segment.thickness_to_chord    = thick_cord	# Thick_cord stored for use in airfoil, below.		
+		segment_root_chord            = vsp.GetParmVal(wing_id, 'Root_Chord', 'XSec_' + str(i)) * units_factor
+		segment.root_chord_percent    = segment_root_chord / total_chord		
+		segment.percent_span_location = proj_span_sum / (total_proj_span/2)
+		segment.twist                 = vsp.GetParmVal(wing_id, 'Twist', 'XSec_' + str(i-1)) * Units.deg
+		
+		if i==start:
+			wing.thickness_to_chord = thick_cord
+	
+		if i < segment_num:      # This excludes the tip xsec, but we need a segment in SUAVE to store airfoil.
+			sweep     = vsp.GetParmVal(wing_id, 'Sweep', 'XSec_' + str(i)) * Units.deg
+			sweep_loc = vsp.GetParmVal(wing_id, 'Sweep_Location', 'XSec_' + str(i))
+			AR        = vsp.GetParmVal(wing_id, 'Aspect', 'XSec_' + str(i))
+			taper     = vsp.GetParmVal(wing_id, 'Taper', 'XSec_' + str(i))
+			   
+			segment_sweeps_quarter_chord[i] = convert_sweep(sweep,sweep_loc,0.25,AR,taper)
+			segment.sweeps.quarter_chord    = segment_sweeps_quarter_chord[i]  # Used again, below
+			
+			# Used for dihedral computation, below.
+			segment_dihedral[i]	      = vsp.GetParmVal(wing_id, 'Dihedral', 'XSec_' + str(i)) * Units.deg 
+			segment.dihedral_outboard     = segment_dihedral[i]
+	
+			segment_spans[i] 	      = vsp.GetParmVal(wing_id, 'Span', 'XSec_' + str(i)) * units_factor
+			proj_span_sum += segment_spans[i] * np.cos(segment_dihedral[i])	
+			span_sum      += segment_spans[i]
+		else:
+			segment.root_chord_percent    = (vsp.GetParmVal(wing_id, 'Tip_Chord', 'XSec_' + str(i-1))) * units_factor /total_chord
+	
+		# XSec airfoil
+		jj = i-1  # Airfoil index i-1 because VSP airfoils and sections are one index off relative to SUAVE.
+		xsec_id = str(vsp.GetXSec(xsec_surf_id, jj))
+		airfoil = Airfoil()
+		if vsp.GetXSecShape(xsec_id) == 7: 	# XSec shape: NACA 4-series
```

Comment:
Use values like vsp.XS_FOUR_SERIES to reduce dependence on specific vsp versions.

Reply:
Done along with the other shapes